src/libbson/src/bson/bson-string.c

```diff
@@ -472,11 +472,11 @@ bson_strndup (const char *str, /* IN */
 void
 bson_strfreev (char **str) /* IN */
 {
-   int i;
-
    if (str) {
-      for (i = 0; str[i]; i++)
+      for (size_t i = 0u; str[i] != NULL; ++i) {
          bson_free (str[i]);
+      }
```

Comment:
Suggest iterating with a `char **`:
```suggestion
      for (char **ptr = str; *ptr != NULL; ++ptr) {
         bson_free (*ptr);
      }
```

This may avoid a possible unsigned overflow with `i` if `str` has more than `SIZE_MAX` entries.

Reply:
+1 to using pointer-based iteration over indexing (with all the complexities that come with it). Updated accordingly.